File: core/ucp/apps/ucp-api/src/ports/outbound/organization.provider.ts
export const ORGANIZATION_PROVIDER = Symbol('ORGANIZATION_PROVIDER');

export interface IOrganizationProvider {
  createOrganization(
    name: string,
  ): Promise<{ orgId: string; grantSucceeded?: boolean }>;
  deleteOrganization(orgId: string): Promise<void>;
}

File: core/ucp/apps/ucp-api/src/ucp_api/ports/outbound/organization_provider.py
from typing import Protocol, Tuple

class IOrganizationProvider(Protocol):
    async def create_organization(self, name: str) -> Tuple[str, bool]:
        """Creates an organization and returns (org_id, grant_succeeded)"""
        ...
        
    async def delete_organization(self, org_id: str) -> None:
        """Deletes an organization"""
        ...
